test/unit/org/apache/cassandra/service/accord/AccordCommandStoreFuzzTest.java

```diff
@@ -0,0 +1,457 @@
+import accord.utils.async.AsyncChains;
+import accord.utils.async.AsyncResult;
+import org.apache.cassandra.ServerTestUtils;
+import org.apache.cassandra.concurrent.ExecutorFactory;
+import org.apache.cassandra.concurrent.ExecutorPlus;
+import org.apache.cassandra.concurrent.ScheduledExecutorPlus;
+import org.apache.cassandra.concurrent.SimulatedExecutorFactory;
+import org.apache.cassandra.concurrent.Stage;
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.marshal.Int32Type;
+import org.apache.cassandra.metrics.AccordStateCacheMetrics;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.SchemaConstants;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.service.accord.api.PartitionKey;
+import org.apache.cassandra.tcm.ClusterMetadata;
+import org.apache.cassandra.tcm.transformations.AddAccordTable;
+import org.apache.cassandra.utils.Generators;
+import org.apache.cassandra.utils.Pair;
+import org.assertj.core.api.Assertions;
+
+import static accord.utils.Property.qt;
+import static org.apache.cassandra.service.accord.AccordTestUtils.createTxn;
+import static org.apache.cassandra.service.accord.AccordTestUtils.wrapInTxn;
+import static org.apache.cassandra.utils.AccordGenerators.fromQT;
+
+public class AccordCommandStoreFuzzTest extends CQLTester
+{
+    static
+    {
+        CassandraRelevantProperties.TEST_ACCORD_STORE_THREAD_CHECKS_ENABLED.setBoolean(false);
+        // Restarts are not covered in these tests and the range logic is getting rewritten, so disable loading
+        CassandraRelevantProperties.TEST_ACCORD_STORE_LOAD_RANGES_ENABLED.setBoolean(false);
+        // since this test does frequent truncates, the info table gets updated and forced flushed... which is 90% of the cost of this test...
+        // this flag disables that flush
+        CassandraRelevantProperties.UNSAFE_SYSTEM.setBoolean(true);
+    }
+
+    private static TableMetadata intTbl, reverseTokenTbl;
+    private static Node.Id nodeId;
+
+    @BeforeClass
+    public static void setUpClass()
+    {
+        CQLTester.setUpClass();
+        DatabaseDescriptor.setIncrementalBackupsEnabled(false);
+    }
+
+    @Before
+    public void init()
+    {
+        if (intTbl != null)
+            return;
+        createKeyspace("CREATE KEYSPACE test WITH replication={ 'class' : 'SimpleStrategy', 'replication_factor' : 2 }");
+        createTable("test", "CREATE TABLE test.tbl1 (pk int PRIMARY KEY, value int)");
+        intTbl = Schema.instance.getTableMetadata("test", "tbl1");
+        AddAccordTable.addTable(intTbl.id);
+
+        createTable("test", "CREATE TABLE test.tbl2 (pk vector<bigint, 2> PRIMARY KEY, value int)");
+        reverseTokenTbl = Schema.instance.getTableMetadata("test", "tbl2");
```

Comment:
ill remove this... this is what the range tests use.  The range tests work with random tokens so can't use `int` as the primary key...